node/lib/util/rm.js

```diff
@@ -194,8 +194,13 @@ const checkCleanliness = co.wrap(function *(repo, headTree, index, pathname,
         return yield checkIndexIsHeadOrWorkdir(repo, headTree, entry,
                                                pathname, displayPath);
     }
-
-    const status = yield NodeGit.Status.file(repo, pathname);
+    let status;
+    try {
+        // Status.file throws errors after 0.22.0
```

Comment:
What did it do before that? And under what condition does it throw?

Reply:
If understand correctly, it expose all errors from lib_git: https://github.com/nodegit/nodegit/pull/1468/files

> GIT_ENOTFOUND if the file is not found in the HEAD, index, and work tree, GIT_EAMBIGUOUS if `path` matches multiple files or if it refers to a folder, and -1 on other errors

Reply:
So, why do we want to ignore these errors?  Shouldn't we tell the user "hey, you tried to delete a file that doesn't exist"?

Reply:
1. This function checks the cleanness instead of deleting a file, as stated in the comment, `An already-deleted file is always clean`. So I think it makes sense to swallow file non-exist error.
2. Main purpose of this pr is to upgrade NodeGit, IMHO,  behavioral changes should be addressed in separated pull requests.

Reply:
OK, I guess let's leave it, since this does seem to make the tests pass, and I did a fair amount of testing on this code.